processor/routingprocessor/router.go

```diff
@@ -34,69 +34,71 @@ import (
 // structure (plog.Logs, pmetric.Metrics and ptrace.Traces respectively) in order
 // to not cause higher CPU usage in the exporters when exproting data (it's always
 // better to batch before exporting).
-type router struct {
+type router[E component.Exporter] struct {
```

Comment:
I wanted to introduce a type constraint for exporters to limit type params only to `component.TracesExporter`,
`component.MetricsExporter`, and `component.LogsExporter` but it seems that currently it is not possible to union types with behavioral interfaces.